compressor_types.go

```diff
@@ -1,21 +1,24 @@
 package compressor
 
 import (
+	"github.com/open-cluster-management/hub-of-hubs-message-compression/compressors"
+	noop "github.com/open-cluster-management/hub-of-hubs-message-compression/compressors/default"
 	"github.com/open-cluster-management/hub-of-hubs-message-compression/compressors/gzip"
 )
 
-// CompressType is the compressing supported types.
+// CompressionType is the compressing supported types.
 //
-// Supported types: GZip.
-type CompressType string
+// Supported types: Default (no op), GZip.
+type CompressionType string
 
 const (
+	// Default returns a no-op compressor (does nothing).
+	Default CompressionType = "default"
 	// GZip is used to create a gzip-based Compressor.
-	GZip CompressType = "gzip"
+	GZip CompressionType = "gzip"
 )
 
-var compressorsMap = map[CompressType]func() Compressor{
-	GZip: func() Compressor {
-		return gzip.NewGzipCompressor()
-	},
```

Comment:
the more I read this file, the more I'm getting convinced that this file is not needed.
considering the fact that `compressor.go` and `types.go` are so small files, both in the same package and `compressor.go` uses `types.go` - why don't we put it in the same file?
I tried to do this change in my local env and I think it makes perfect sense.. let me know what you think.
for example - having the following `compressor.go` file:
```
// CompressionType the supported compression types.
type CompressionType string

const (
	// NoOp used to create a no-op compressor (does nothing).
	NoOp CompressionType = "no-op"
	// GZip used to create a gzip-based Compressor.
	GZip CompressionType = "gzip"
)

var errCompressionTypeNotFound = errors.New("compression type not supported")

// NewCompressor returns a compressor instance that corresponds to the given CompressorType.
func NewCompressor(compressionType CompressionType) (compressors.Compressor, error) {
	supportedCompressors := map[CompressionType]func() compressors.Compressor{
		NoOp: noop.NewNoOpCompressor,
		GZip: gzip.NewGZipCompressor,
	}

	createCompressorFunc, found := supportedCompressors[compressionType]
	if !found {
		return nil, errCompressionTypeNotFound
	}

	return createCompressorFunc(), nil
}
```